jsapp/js/components/submissions/submissionUtils.ts

```diff
@@ -12,35 +12,49 @@ import {
   GROUP_TYPES_BEGIN,
   QUESTION_TYPES,
   CHOICE_LISTS,
+  AnyRowTypeName,
 } from 'js/constants';
 
+enum DisplayGroupTypeName {
+  group_root = 'group_root',
+  group_repeat = 'group_repeat',
+  group_regular = 'group_regular',
+  group_matrix = 'group_matrix',
+  group_matrix_row = 'group_matrix_row',
+}
+
 export const DISPLAY_GROUP_TYPES = createEnum([
-  'group_root',
-  'group_repeat',
-  'group_regular',
-  'group_matrix',
-  'group_matrix_row',
-]);
+  DisplayGroupTypeName.group_root,
+  DisplayGroupTypeName.group_repeat,
+  DisplayGroupTypeName.group_regular,
+  DisplayGroupTypeName.group_matrix,
+  DisplayGroupTypeName.group_matrix_row,
+]) as {[P in DisplayGroupTypeName]: DisplayGroupTypeName};;
 
-/**
- * @typedef {Object} DisplayGroup
- * @property {string} type - One of DISPLAY_GROUP_TYPES
- * @property {string} label - Localized display label
- * @property {string} name - Unique identifier
- * @property {Array<DisplayResponse|DisplayGroup>} children - List of groups and responses
- */
 class DisplayGroup {
-  constructor(type, label = null, name = null) {
+  public type: DisplayGroupTypeName;
+  /** Localized display label */
+  public label: string | null = null;
+  /** Unique identifier */
+  public name: string | null = null;
+  /** List of groups and responses */
+  public children: Array<DisplayResponse|DisplayGroup> = [];
```

Comment:
Just curious of the syntactic difference between this and `(DisplayResponse | DisplayGroup)[]`?

Reply:
So I found this post from 2017 https://mattferderer.com/difference-between-array-types-in-typescript after a quick look. I didn't find any good article that covers all of the ways, but I think this is the same :)